### ISSUE TITLE
Documentation for Python container classes

### ISSUE THREAD

**Issue by magimagi** [sphinx-doc/sphinx#3319]
Subject: Documentation for Python container classes- only the first inner class title doesn't include the outer class name.

### Problem
while trying to build a documentation for a container class that includes more classes inside it.
e.g. 
```
class A (object):
    """ Container Class"
     
     class B(object):
     "Inner class 1"

     class C(object):
     "Inner class 2"
```
When using the autoclass  for the container class (A) with the members option, as followed:
```
 .. autoclass:: A
   :members:
   :member-order: bysource
```

sphinx will document the title of the inner classes, but will exclude the outer class name only from the first inner class.

the result will look like this:
```
Class A
 Container Class

      Class B
      Inner class 1
      
      Class A.C
      Inner class 2
```
I added : `add_module_names = False`  in the conf.py file. 

what I want to get is either no outer class names in the inner class titles, or that all of them will contain the outer class name...

#### Procedure to reproduce the problem
```
<Paste your command-line here which cause the problem>
```

#### Error logs / results
```
<Paste your error log here>
```
- <public link of unexpected result if you have>

#### Expected results
<Describe what to actually do>

### Reproducible project / your project
- <link to your project, or attach zipped small project sample>

### Environment info
- OS: Linux
- Python version:  2.7
- Sphinx version: latests 
- <Extra tools e.g.: Browser, tex or something else>

**Comment by magimagi** [comment]
Issue appears with both: add_module_names = False/True.

**Comment by tk0miya** [comment]
Note:
Internally, autodoc generates following reST code on building docs:
```
.. py:class:: A
   :module: mod

   Container Class


   .. py:class:: A.B
      :module: mod

      Inner class 1


   .. py:class:: A.C
      :module: mod

      Inner class 2
```
But python domain fails to render it correctly.
As a trial, I marked up the reST above directly into my index.rst (without autodoc module). And I got the same problem.

So this is a bug of python domain.

**Comment by tk0miya** [comment]
refs: #3065

**Comment by tk0miya** [comment]
It seems big changes are needed to fix this (refs: #3067).
So I move the mileston to 1.6 release.

**Comment by tk0miya** [comment]
Fixed by #3520.
Finally, it will be fixed in 1.5.4.
